express/component_spec.py

```diff
@@ -0,0 +1,116 @@
+"""This module defines classes to represent an Express component specifications."""
+import json
+import pkgutil
+import yaml
+import typing as t
+from typing import Dict
+from dataclasses import dataclass
+
+import jsonschema.exceptions
+from jsonschema import Draft4Validator
+
+from express.exceptions import InvalidComponentSpec
+
+
+def load_yaml(yaml_path: str) -> Dict:
+    """Loads a YAML file and returns a dictionary."""
+    try:
+        with open(yaml_path) as f:
+            return yaml.safe_load(f)
+    except Exception as e:
+        raise ValueError(f"Failed to load YAML file: {e}")
+
+
+@dataclass
+class ComponentField:
+    """Class representing a single field or column in an Express component subset."""
+
+    name: str
+    type: str
+
+
+@dataclass
+class ComponentSubsets:
```

Comment:
```suggestion
class ComponentSubset:
```
Looks like this class represents a single subset